### PR TITLE
Feat: Implement dorq wf logs command

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

User needs to gather logs for entire workflow

# This PR's solution

Create new CLI command - dorq wf logs 

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

### CODE REVIEW

File: src/orquestra/sdk/_base/cli/_dorq/_entry.py [hunk 1]
```diff
@@ -15,6 +15,26 @@
...
+)
+
+CONFIG_OPTION = cloup.option(
+    "-c",
+    "--config",
+    required=False,
+    help="""
+Name of the config used to submit workflow. Use 'in-process' for running workflow
+as local python process, 'ray' to run workflow in local ray cluster.
+To get confing name for remote runtime, use orq login -s <uri> first
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/cli/_dorq/_entry.py:34]:
```suggestion
To get config name for remote runtime, use orq login -s <uri> first.
```

File: src/orquestra/sdk/_base/cli/_dorq/_entry.py [hunk 2]
```diff
@@ -142,7 +150,30 @@ def results(
...
-@cloup.option("-c", "--config")
+@CONFIG_OPTION
+@DOWNLOAD_DIR_OPTION
+def logs(
+    wf_run_id: t.Optional[str],
+    config: t.Optional[str],
+    download_dir: t.Optional[Path],
+):
+    """
+    Shows logs gathered during execution of a workflow.
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/cli/_dorq/_entry.py:161]:
```suggestion
    Shows logs gathered during execution of a workflow produced by all tasks.
```

File: tests/cli/dorq/test_dumpers.py [hunk 3]
```diff
@@ -73,3 +73,38 @@ def test_no_dir(tmp_path: Path):
...
+
+class TestLogsDumper:
+    @staticmethod
+    def test_logs_dumper(tmp_path: Path):
+        # Given
+        log_values = ["my_logs", "next_log"]
+        task_invocation = "my_task_invocation"
+        logs = {task_invocation: log_values}
+        wf_run_id = "wf.1234"
+        wf_log_file = wf_run_id + ".logs"
```
Comment (alexjuda) [inline tests/cli/dorq/test_dumpers.py:86]: **suggestion**: I think `.log` is a more idiomatic file extension than `.logs`
```suggestion
        wf_log_file = wf_run_id + ".log"
```

File: tests/cli/dorq/workflow/test_logs.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            wf_run_id = "<wf run ID sentinel>"
+            config = "<config sentinel>"
+            download_dir = None
+
+            # Resolved values
+            resolved_id = "<resolved ID>"
+            resolved_config = "<resolved config>"
+
+            # Mocks
+            presenter = create_autospec(WrappedCorqOutputPresenter)
```
Comment (alexjuda) [inline tests/cli/dorq/workflow/test_logs.py:48]: **question**: would you recommend using `create_autospec` everywhere we would use a `Mock()`?

**Review by alexjuda — APPROVED** [review]
Approving but please take a look at my suggestions before merging!

**Comment by alexjuda** [comment]
Also, please fill the changelog entry.

File: tests/cli/dorq/workflow/test_logs.py (hunk 4 above)
Comment (SebastianMorawiec) [reply tests/cli/dorq/workflow/test_logs.py:48]: Explained on slack why we found it to be better in those particular cases where we stub entire interface/class

**Pull request merged**